mitodl/odl-video-service
# Added request to auto-add HLS videos to edX when appropriate

#### What are the relevant tickets?
Closes #762 

#### What's this PR do?
When a video is added to a collection and successfully transcoded, this PR adds an API request to add the HLS-encoded video to edX (triggered automatically when the model object is created)

#### How should this be manually tested?
- Run devstack with the `edx-video-api` package installed (details in the [project README](https://github.com/mitodl/edx-video-api))
- Add settings to your `.env`:
    ```
    EDX_API_KEY=PUT_YOUR_API_KEY_HERE
    EDX_ACCESS_TOKEN=<access token value for some admin user in edX>
    # OSX users
    EDX_BASE_URL=http://host.docker.internal:18000
    # Linux users
    EDX_BASE_URL=http://localhost:18000
    ```
- Set the `edx_course_id` value for some collection to a course id in your LMS instance (e.g.: `course-v1:edX+DemoX+Demo_Course`). This can be done in Django admin
- Add a video to that collection

When the video is done uploading and transcoding, a successful request should be made to add the HLS-encoded video to edX. You should see that request in the celery logs.

#### Any background context you want to provide?
I added the necessary config files to allow devs to run a container-ized Jupyter notebook. I added it so I could do some of my own data manipulation and it made enough sense just to leave those files in the PR
